docs/reference/mount.md

```diff
@@ -52,6 +52,58 @@ Searching for lakeFS credentials and server endpoint in the following order:
 - `LAKECTL_*` Environment variables
 - `~/.lakectl.yaml` Configuration file or via `--lakectl-config` flag
 
+## Authenticating with lakeFS (using AWS IAM Role)
+
+Starting from **lakeFS ≥ v1.57.0** and **Everest ≥ v0.4.0**, authenticating with IAM roles is supported!  
+When IAM authentication is configured, Everest will use your machine's **default AWS profile credentials** to generate a **session token** used for authenticating against lakeFS.  
+This token is seamlessly refreshed as long as the AWS session remains valid.  
+For more information regarding IAM authentication, please check [external principals aws](https://docs.lakefs.io/security/external-principals-aws.html).
+
+Because IAM roles are linked to users, lakeFS must map each role to a user.  
+To do this, make sure the IAM role is attached to lakeFS. See [Administration of IAM Roles in lakeFS](https://docs.lakefs.io/security/external-principals-aws.html#administration-of-iam-roles-in-lakefs)
+
+### Authentication Chain
+
+When running an Everest `mount` command, authentication occurs in the following order:
+
+1. **Session token** from the environment variable `EVEREST_LAKEFS_CREDENTIALS_SESSION_TOKEN` or `LAKECTL_CREDENTIALS_SESSION_TOKEN`.  
+   If the token is expired, authentication will fail.
+2. **lakeFS key pair**, using lakeFS access key ID and secret key. (picked up from lakectl if Everest not provided)
+3. **IAM authentication**, if configured.
+
+### Configuration
+
+Since IAM authentication is used as a fallback method, `.lakectl.yaml` **must not contain static credentials**.
+
+To use IAM authentication, new configuration fields were introduced:
+
+- `CREDENTIALS.PROVIDER.TYPE = aws_iam`: The identity provider type. Must be equal to `aws_iam`.
+- `CREDENTIALS.PROVIDER.AWS_IAM.TOKEN_TTL_SECONDS`: Session token's validation duration.  
+- `CREDENTIALS.PROVIDER.AWS_IAM.URL_PRESIGN_TTL_SECONDS`: AWS STS's presigned URL validation duration.  
+- `CREDENTIALS.PROVIDER.AWS_IAM.REFRESH_INTERVAL`: The amount of time before token expiration that Everest will try to fetch a new session token instead of using the current one.  
+- `CREDENTIALS.PROVIDER.AWS_IAM.TOKEN_REQUEST_HEADERS`: Map of required `<header>:<value>` to be signed by the AWS STS request.  
+- `CREDENTIALS.PROVIDER.AWS_IAM.TOKEN_REQUEST_HEADERS: x-lakefs-server-id:<lakeFS host>`: Required header.
+
+These configuration fields can be set via `.lakectl.yaml`: 
+
+```yaml
+credentials:
+  provider:
+    type: aws_iam          # Required
+    aws_iam:
+      token_ttl_seconds: 60m              # Optional, default: 1h
+      url_presign_ttl_seconds: 15m        # Optional, default: 15m
+      refresh_interval: 5m                # Optional, default: 5m
+      token_request_headers:              # Required
+        x-lakefs-server-id: <lakeFS host>     # Replace with your actual lakeFS host
```

Comment:
This is confusing since by default it adds this
```suggestion
        # x-lakefs-server-id: <lakeFS host>     Added by default if token_request_headers is not set
```